apps/salesforce-commerce-cloud/src/App.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { locations } from '@contentful/app-sdk';
+import { useSDK } from '@contentful/react-apps-toolkit';
+import { useMemo } from 'react';
+import ConfigScreen from './locations/ConfigScreen';
+import Dialog from './locations/Dialog';
+import Field from './locations/Field';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+
+const ComponentLocationSettings = {
+  [locations.LOCATION_APP_CONFIG]: ConfigScreen,
+  [locations.LOCATION_ENTRY_FIELD]: Field,
+  [locations.LOCATION_DIALOG]: Dialog,
+};
+
+const queryClient = new QueryClient();
+
+const App = () => {
+  const sdk = useSDK();
+
+  const Component = useMemo(() => {
+    for (const [location, component] of Object.entries(ComponentLocationSettings)) {
+      if (sdk.location.is(location)) {
+        return component;
+      }
+    }
+  }, [sdk.location]);
```

Comment:
This looks like this may be better suited for a useEffect, since we want this hooked into the react lifecycle

```suggestion
  const [Component, setComponent] = useState<ReactElement>();
  
  useEffect(() => {
    for (const [location, component] of Object.entries(ComponentLocationSettings)) {
      if (sdk.location.is(location)) {
        setComponent(component)
      }
    }
  }, [sdk.location]);
```